frontend/src/js/components/deployments/deployment-wizard/SoftwareDevices.tsx

```diff
@@ -300,18 +282,25 @@ export const Software = ({ commonClasses, deploymentObject, releaseRef, releases
           {releaseSelectionLocked ? (
             <TextField value={deploymentRelease?.name} label="Release" disabled className={classes.infoStyle} />
           ) : (
-            <AsyncAutocomplete
-              id="deployment-release-selection"
-              initialValue={deploymentRelease?.name}
-              labelAttribute="name"
-              placeholder="Select a Release"
-              selectionAttribute="name"
-              options={releaseItems}
-              onChange={onReleaseInputChange}
-              onChangeSelection={onReleaseSelectionChange}
-              isLoading={isLoadingReleases}
-              styles={hardCodedStyle}
-            />
+            <>
+              <ReleaseArtifactFilter
+                device={device}
+                releases={releaseItems}
+                onSelect={onReleaseSelectionChange}
+                open={releaseFilterOpened}
+                onClose={() => setReleaseFilterOpened(false)}
+              />
+              <Button
+                size="large"
+                color="neutral"
+                variant="outlined"
+                className={`text-overflow ${classes.releaseSelect}`}
+                endIcon={releaseFilterOpened ? <ExpandLessIcon /> : <ExpandMoreIcon />}
+                onClick={() => setReleaseFilterOpened(!releaseFilterOpened)}
+              >
+                <span className={`${classes.releaseSelectText} text-overflow`}>{deploymentRelease?.name ?? 'Select a release'}</span>
```

Comment:
I haven't looked into it, but the double `text-overflow` here looks a bit suspicious...